core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -371,6 +374,10 @@ public PartitionSpec spec(int id) {
     return specsById;
   }
 
+  int lastAssignedFieldId() {
```

Comment:
I think this name is potentially confusing. It is the last assigned partition field ID, and "partition" is a key part of it. How about renaming it to `lastAssignedPartitionId` instead?

Reply:
Updated the field name in `TableMetadata`. 
In `PartitionSpec`, there is also a `lastAssignedFieldId` field. It seems to be fine as it is a field inside partition spec, which already indicates it is about partition.